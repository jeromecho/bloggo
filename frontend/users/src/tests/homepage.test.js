const React = require("react");
// import { ... } from  '@testing-library/react';
const userEvent = require('@testing-library/user-event');
const { Home } = require('../pages/Home');
const renderer = require('react-test-renderer')
const { MemoryRouter } = require('react-router-dom');

it ("matches snapshot", () => {
    const tree = renderer
        .create(<MemoryRouter><Home /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
