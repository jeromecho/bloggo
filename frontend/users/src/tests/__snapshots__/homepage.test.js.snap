// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <header>
    <div>
      <div
        className="left"
      >
        <img
          src=""
        />
        <h2>
          BLOGGO
        </h2>
      </div>
      <div
        className="right"
      >
        <div
          className="switch-container"
        >
          <img
            src=""
          />
          <div
            style={
              Object {
                "MozTransition": "opacity 0.25s",
                "MozUserSelect": "none",
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "WebkitTransition": "opacity 0.25s",
                "WebkitUserSelect": "none",
                "borderRadius": 10,
                "direction": "ltr",
                "display": "inline-block",
                "msUserSelect": "none",
                "opacity": 1,
                "position": "relative",
                "textAlign": "left",
                "touchAction": "none",
                "transition": "opacity 0.25s",
                "userSelect": "none",
              }
            }
          >
            <div
              className="react-switch-bg"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "MozTransition": "background 0.25s",
                  "WebkitTransition": "background 0.25s",
                  "background": "#301414",
                  "borderRadius": 10,
                  "cursor": "pointer",
                  "height": 20,
                  "margin": 0,
                  "position": "relative",
                  "transition": "background 0.25s",
                  "width": 50,
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransition": "opacity 0.25s",
                    "WebkitTransition": "opacity 0.25s",
                    "height": 20,
                    "opacity": 0,
                    "pointerEvents": "none",
                    "position": "relative",
                    "transition": "opacity 0.25s",
                    "width": 30,
                  }
                }
              >
                <svg
                  height="100%"
                  style={
                    Object {
                      "position": "absolute",
                      "top": 0,
                    }
                  }
                  viewBox="-2 -5 17 21"
                  width="100%"
                >
                  <path
                    d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                    fill="#fff"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <div
                style={
                  Object {
                    "MozTransition": "opacity 0.25s",
                    "WebkitTransition": "opacity 0.25s",
                    "height": 20,
                    "opacity": 1,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transition": "opacity 0.25s",
                    "width": 30,
                  }
                }
              >
                <svg
                  height="100%"
                  style={
                    Object {
                      "position": "absolute",
                      "top": 0,
                    }
                  }
                  viewBox="-2 -5 14 20"
                  width="100%"
                >
                  <path
                    d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                    fill="#fff"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              className="react-switch-handle"
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "background": "#FFF8FA",
                  "border": 0,
                  "borderRadius": "50%",
                  "boxShadow": null,
                  "cursor": "pointer",
                  "display": "inline-block",
                  "height": 20,
                  "outline": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateX(0px)",
                  "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "width": 20,
                }
              }
            />
            <input
              aria-checked={false}
              checked={false}
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyUp={[Function]}
              role="switch"
              style={
                Object {
                  "border": 0,
                  "clip": "rect(0 0 0 0)",
                  "height": 1,
                  "margin": -1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "absolute",
                  "width": 1,
                }
              }
              type="checkbox"
            />
          </div>
          <img
            src=""
          />
        </div>
      </div>
    </div>
  </header>,
  <main
    className="home"
  >
    <div
      className="blur"
    />
    <div
      className="spacer"
    />
    <div>
      <div
        className="hero-content"
      >
        <h1>
          Stories & Adventures
        </h1>
        <p>
          Hear of Jerome's latest thoughts and adventures on his personal blog
        </p>
      </div>
      <div
        className="posts-content"
      >
        <div>
          <h3>
            All Posts
          </h3>
          <hr />
        </div>
        <div
          className="posts"
        >
          <div
            className="post"
          >
            <div
              className="left"
            >
              <h4>
                Not Connected to Ser...
              </h4>
              <p>
                2022-09-09
              </p>
              <p>
                Tim
              </p>
            </div>
            <div
              className="right"
            >
              <a
                href="/default id"
                onClick={[Function]}
              >
                View Post
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="spacer"
    />
    <div
      className="blur"
    />
  </main>,
  <footer>
    <div
      id="footer-content-container"
    >
      <p>
        Made with love by Cho Industries    -     ©2022
      </p>
    </div>
  </footer>,
]
`;
